src/nvim/api/vim.c

```diff
@@ -1041,12 +1042,33 @@ Integer nvim_open_term(Buffer buffer, Dict(open_term) *opts, Error *err)
     .close_cb = term_close,
     .force_crlf = GET_BOOL_OR_TRUE(opts, open_term, force_crlf),
   };
+
+  // Read existing buffer contents (if any)
+  StringBuilder contents = KV_INITIAL_VALUE;
+  for (linenr_T lnum = 1; lnum <= buf->b_ml.ml_line_count; lnum++) {
+    char *bufstr = ml_get_buf(buf, lnum);
+    colnr_T bufstrlen = ml_get_buf_len(buf, lnum);
+    kv_concat_len(contents, bufstr, (size_t)bufstrlen);
```

Comment:
Translate NL to Nul before concatenating?

Reply:
Good suggestion, I refactored that code out and re-used it.